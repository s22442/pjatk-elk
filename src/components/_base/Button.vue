<template>
  <button
    _rounded
    _inline-block
    _p="x4 y.5"
    _fw600
    _bg-accent-interactive
    _text-accent-interactive-contrast
    _transition-colors
    _cursor-pointer
    _select-none
    _disabled:pointer-events-none
  >
    <slot />
  </button>
</template>
