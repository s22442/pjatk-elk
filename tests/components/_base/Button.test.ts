import Button from '~/components/_base/Button.vue';

import { mount } from '@vue/test-utils';

describe('Button', () => {
  it('should render', () => {
    const wrapper = mount(Button, { slots: { default: 'abc' } });
    expect(wrapper.text()).toContain('abc');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
