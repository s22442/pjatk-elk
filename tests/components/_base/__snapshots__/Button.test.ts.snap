// Vitest Snapshot v1

exports[`Button > should render 1`] = `"<button _rounded=\\"\\" _inline-block=\\"\\" _p=\\"x4 y.5\\" _fw600=\\"\\" _bg-accent-interactive=\\"\\" _text-accent-interactive-contrast=\\"\\" _transition-colors=\\"\\" _cursor-pointer=\\"\\" _select-none=\\"\\" _disabled:pointer-events-none=\\"\\">abc</button>"`;
